_stbt/core.py

```diff
@@ -1005,21 +1005,26 @@ def _callable_description(callable_):
     '    lambda: stbt.press("OK"))\\n'
     >>> _callable_description(functools.partial(int, base=2))
     'int'
+    >>> _callable_description(functools.partial(functools.partial(int, base=2),
+    ...                                         x='10'))
+    'int'
+    >>> class T(object):
+    ...     def __call__(self): return True;
+    >>> _callable_description(T())
+    '<_stbt.core.T object at 0x...>'
     """
-    try:
-        d = callable_.__name__
-    except AttributeError:
-        if isinstance(callable_, functools.partial):
-            # functools.partial wraps the original function
-            d = callable_.func.__name__
-        else:
-            raise
-    if d == "<lambda>":
-        try:
-            d = inspect.getsource(callable_)
-        except IOError:
-            pass
-    return d
+    if hasattr(callable_, "__name__"):
+        name = callable_.__name__
```

Comment:
Not important but I tend to prefer doing something and then checking if it succeeded to checking whether it will succeed before doing it.  It's not important in this instance but it can be good for avoiding race-conditions.

If you wanted to avoid `try...except` `getattr` could be used:

    name = getattr(callable_, __name__, None)